Source/Scene/Scene.js

```diff
@@ -1643,76 +1649,162 @@ define([
+        if (scene._useWebVR && mode !== SceneMode.SCENE2D) {
+            updatePrimitives(scene);
+            createPotentiallyVisibleSet(scene);
+            updateAndClearFramebuffers(scene, passState, backgroundColor, picking);
+            executeComputeCommands(scene);
 
-                viewport.x = 0;
-                viewport.y = 0;
-                viewport.width = context.drawingBufferWidth * 0.5;
-                viewport.height = context.drawingBufferHeight;
+            // Based on Calculating Stereo pairs by Paul Bourke
+            // http://paulbourke.net/stereographics/stereorender/
 
-                var savedCamera = Camera.clone(camera, scene._cameraVR);
+            viewport.x = 0;
+            viewport.y = 0;
+            viewport.width = context.drawingBufferWidth * 0.5;
+            viewport.height = context.drawingBufferHeight;
 
-                var near = camera.frustum.near;
-                var fo = near * 5.0;
-                var eyeSeparation = fo / 30.0;
-                var eyeTranslation = Cartesian3.multiplyByScalar(savedCamera.right, eyeSeparation * 0.5, scratchEyeTranslation);
+            var savedCamera = Camera.clone(camera, scene._cameraVR);
 
-                camera.frustum.aspectRatio = viewport.width / viewport.height;
+            var near = camera.frustum.near;
+            var fo = near * 5.0;
+            var eyeSeparation = fo / 30.0;
+            var eyeTranslation = Cartesian3.multiplyByScalar(savedCamera.right, eyeSeparation * 0.5, scratchEyeTranslation);
 
-                var offset = 0.5 * eyeSeparation * near / fo;
+            camera.frustum.aspectRatio = viewport.width / viewport.height;
 
-                Cartesian3.add(savedCamera.position, eyeTranslation, camera.position);
-                camera.frustum.xOffset = offset;
+            var offset = 0.5 * eyeSeparation * near / fo;
 
-                executeCommands(scene, passState);
+            Cartesian3.add(savedCamera.position, eyeTranslation, camera.position);
+            camera.frustum.xOffset = offset;
 
-                viewport.x = passState.viewport.width;
+            executeCommands(scene, passState);
 
-                Cartesian3.subtract(savedCamera.position, eyeTranslation, camera.position);
-                camera.frustum.xOffset = -offset;
+            viewport.x = passState.viewport.width;
 
-                executeCommands(scene, passState);
+            Cartesian3.subtract(savedCamera.position, eyeTranslation, camera.position);
+            camera.frustum.xOffset = -offset;
 
-                Camera.clone(savedCamera, camera);
+            executeCommands(scene, passState);
+
+            Camera.clone(savedCamera, camera);
+        } else {
+            viewport.x = 0;
+            viewport.y = 0;
+            viewport.width = context.drawingBufferWidth;
+            viewport.height = context.drawingBufferHeight;
+
+            if (mode !== SceneMode.SCENE2D) {
+                updatePrimitives(scene);
```

Comment:
This series of function calls is duplicated several times below.  Should we put them in a separate function?